python/pyspark/errors/error_classes.py

```diff
@@ -269,6 +269,11 @@
       "<objectName> is not iterable."
     ]
   },
+  "NOT_LIST" : {
```

Comment:
We should consolidate these `NOT_xxx` to one error as they all throw `xx should be but got yy`?

Reply:
Yes, we can classify error classes starting with `NOT_XXX` into one error class or group similar types of errors into a main-sub error class structure. Each way has its pros and cons, so we plan to proceed with the initial migration in the current form first, and then gather opinions from ppl through decision docs or something before making changes all at once.

BTW, we already have a [ticket](https://issues.apache.org/jira/browse/SPARK-43020) for this FYI :-)

Reply:
Sounds good then!